### DISCUSSION TITLE
Performance issues with ComboBox for big lists(100+ data)

### DISCUSSION THREAD

**Discussion by banushanv** [tailwindlabs/headlessui#2984] [Help]
I am trying to populate a big list (200 elements) into the comboBox headless UI component found in Vue 3, TailwindUI for my project with latest versions. generating the list can sometimes be slow as well.
when I click comboBox input getting delay to open dropdown

**Comment by ncphillips** [comment]
I am having the exact same issue.

We sometimes have thousands of elements in our combobox. I'm on an M2 mac and the delay is 6 seconds. I can't imagine how bad it is for people with slower machines.

I hoped there was some kind of lazy-rendering option but I don't see anything. This might force us to use a different approach entirely.

**Reply by vittxr** [reply]
it's possible use [react-window](https://github.com/bvaughn/react-window) for lazy-rendering, but it would be great if the component could handle it by itself.

**Reply by ncphillips** [reply]
Nope, we're using Vue

**Comment by vittxr** [comment]
I am having the same issue too. I tested it with an `array.length = 3000` and the application crashes in Chrome 122v and in Opera 107v

if someone need to test, following is attached a simple react-app reproducing the error: 

[test.zip](https://github.com/tailwindlabs/headlessui/files/14486217/test.zip)

**Comment by Plinpod** [comment]
Same identical issue.

**Comment by ben519** [comment]
In my case, [virtual scrolling](https://headlessui.com/react/combobox#virtual-scrolling) gave a surprising performance boost. I don't understand why my _unfiltered_ list renders quicker with virtual scrolling enabled, but it does 🤷‍♂️
